File: April/Dhruvabhat24/Java/April 15/Solution.java
class Solution {
    public int sumTree(TreeNode root,int sum){
        if(root==null) return 0;
        sum=sum*10+root.val;
        if(root.left==null && root.right==null) return sum;
        return sumTree(root.left,sum)+sumTree(root.right,sum);
    }
    public int sumNumbers(TreeNode root) {
                return sumTree(root,0);
    }

}

File: April/Dhruvabhat24/C++/April 15/Solution.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int sumNumbers(TreeNode* root) {
        int ans = 0;
        
        function<void(TreeNode*, int)> dfs = [&](TreeNode* node, int path) {
            if (!node) return;
            if (!node->left && !node->right) {
                ans += path * 10 + node->val;
                return;
            }
            dfs(node->left, path * 10 + node->val);
            dfs(node->right, path * 10 + node->val);
        };
        
        dfs(root, 0);
        return ans;
    }
};
